Sources/NIOSOCKS/Channel Handlers/SOCKSServerHandshakeHandler.swift

```diff
@@ -91,6 +91,12 @@ public final class SOCKSServerHandshakeHandler: ChannelDuplexHandler, RemovableC
         var buffer = context.channel.allocator.buffer(capacity: 16)
         buffer.writeMethodSelection(method)
         context.write(self.wrapOutboundOut(buffer), promise: promise)
+        
+        // fast path to check if authentication can be marked as complete
+        // only applies when there is no authentication
+        if method.method == .noneRequired {
+            try self.handleWriteAuthenticationData(context.channel.allocator.buffer(capacity: 0), complete: true, context: context, promise: nil)
```

Comment:
I think we can reimagine this a bit better: on `sendAuthenticationMethod` with `noneRequired`, the state machine should simply transition into the expected state. We also need to keep track of the auth method we used so that we can ensure we tolerate zero-sized auth data with complete if we used the right method.